entity-framework/core/managing-schemas/migrations/index.md

```diff
@@ -167,22 +178,19 @@ dotnet ef migrations script
 
 There are several options to this command.
 
-The **from** migration should be the last migration applied to the database before running the script. If no migrations
-have been applied, specify `0` (this is the default).
+The **from** migration should be the last migration applied to the database before running the script. If no migrations have been applied, specify `0` (this is the default).
 
-The **to** migration is the last migration that will be applied to the database after running the script. This defaults
-to the last migration in your project.
+The **to** migration is the last migration that will be applied to the database after running the script. This defaults to the last migration in your project.
 
-An **idempotent** script can optionally be generated. This script only applies migrations if they haven't already been
-applied to the database. This is useful if you don't exactly know what the last migration applied to the database was or
-if you are deploying to multiple databases that may each be at a different migration.
+An **idempotent** script can optionally be generated. This script only applies migrations if they haven't already been applied to the database. This is useful if you don't exactly know what the last migration applied to the database was or if you are deploying to multiple databases that may each be at a different migration.
 
-Applying migrations at runtime
-------------------------------
+Apply migrations at runtime
+---------------------------
 Some apps may want to apply migrations at runtime during startup or first run. Do this using the `Migrate()` method.
 
-Caution, this approach isn't for everyone. While it's great for apps with a local database, most applications will
-require more robust deployment strategy like generating SQL scripts.
+This method builds on top of the `IMigrator` service, which can be used for more advanced scenarios. Use `DbContext.GetService<IMigrator>()` to access it.
+
+Caution, this approach isn't for everyone. While it's great for apps with a local database, most applications will require more robust deployment strategy like generating SQL scripts.
```

Comment:
Suggestion: move this "caution" after the code snippet an also make it another warning. Both are really pits of failure.